accepted/2021/system-drawing-win-only/system-drawing-win-only.md

```diff
@@ -0,0 +1,213 @@
+
+### Non-Goals
+
+* Closing the gap of `System.Drawing.Common` for non-Windows platforms or
+  bundling `libgdiplus` as part of the .NET 6 installer
+
+## Stakeholders and Reviewers
+
+* Libraries team
+* ML.NET
+* Xamarin team
+
+## Design
+
+In the past we decided to include the following disclaimer in our docs:
+
+> The `System.Drawing.Common` NuGet package works on Windows, Linux, and macOS.
+> However, there are some platform differences. On Linux and macOS, the GDI+
+> functionality is implemented by the `libgdiplus` library. This library is not
+> installed by default in most Linux distributions and doesn't support all the
+> functionality of GDI+ on Windows on macOS. There are also platforms where
+> `libgdiplus` is not available at all. To use types from the
+> `System.Drawing.Common` package on Linux and macOS, you must install
+> `libgdiplus` separately. For more information, see Install .NET on Linux or
+> Install .NET on macOS.
+>
+> If you can't use System.Drawing with your application, recommended
+> alternatives include ImageSharp, SkiaSharp, and Windows Imaging Components.
+
+However, this disclaimer doesn't feel clear enough to steer people towards using
+other graphics libraries.
+
+The proposal is to do the following for .NET 6:
+
+1. **Mark the assembly as only being supported on Windows**. That will cause the
+   platform compatibility analyzer to warn when consumed from a project that
+   isn't targeting net6.0-windows (or marked as Windows-specific).
+
+2. Throw `PlatformNotSupportedException` when trying to load `libgdiplus`
+   pointing to an aka.ms link with the reasons behind it and recommending
+   stable, well-maintained open-source library alternatives
+
+3. Add targets to the package that when it is restored using a non-windows RID,
+   it warns that we no longer support it on non-windows and point to the aka.ms
+   link with more info.
+
+The challenge is that `System.Drawing.Common` is used by 1st party consumers
+such as ML.NET, which uses it for image manipulation and even exposes an API
+that returns a Bitmap. At the same time, https://github.com/dotnet/iot is in the
+process of moving to use ImageSharp and so far they had a great migration
+experience. We need to engage with 1st party customers to see what impact this
+proposal will have on them.
+
+***Note:** This only affect `System.Drawing.Common` and would leave
+`System.Drawing.Primitives` as-is. That assembly contains primitive types that
+don't depend on GDI+, such `Rectangle`, `Point`, and `Size`. Any types in
+`System.Drawing.Common` that don't depend on GDI+, will be moved to
+`System.Drawing.Primitives`.*
+
+## Q & A
+
+### Will this prevent applications to move to .NET 6?
+
+No, `System.Drawing.Common` is a standalone package that does not ship as part
+of the shared framework. For that reason if an application wants to keep using
+`System.Drawing.Common` on `Unix` and move to .NET 6, they can reference the
+`5.0.x` version of the package.
```

Comment:
```suggestion
No, `System.Drawing.Common` is a standalone package that does not ship as part
of the shared framework. For that reason if an application wants to keep using
`System.Drawing.Common` on `Unix` and move to .NET 6, they can either pin the version of the package to `5.0.x` or use the `6.0.x` package and set a runtime config switch `System.Drawing.UnixSupport` to `true`. 

However application should be aware that we will not provide any support for Unix issues starting from .NET 6.
```

Reply:
@terrajobst what do you think?

Reply:
I'd remove the pinning as that is a non-standard workflow and some of our packages (i.e. most of ASP.NET Core won't let you do that). I'd just leave the runtime config switch as the primary mitigation.